app/utils/forms.tsx

```diff
@@ -98,25 +99,42 @@ export function CheckboxField({
 	errors,
 }: {
 	labelProps: Omit<JSX.IntrinsicElements['label'], 'className'>
-	buttonProps: Omit<
-		React.ComponentPropsWithoutRef<typeof Checkbox.Root>,
-		'type' | 'className'
-	> & {
-		type?: string
-	}
+	buttonProps: Omit<JSX.IntrinsicElements['input'], 'className'>
 	errors?: ListOfErrors
 }) {
 	const fallbackId = useId()
 	const id = buttonProps.id ?? buttonProps.name ?? fallbackId
 	const errorId = errors?.length ? `${id}-error` : undefined
+	// To emulate naitve input events that Conform listen to:
+	// See https://conform.guide/integrations
+	const [ref, control] = useInputEvent()
+	const buttonRef = useRef<HTMLButtonElement>(null)
 	return (
 		<div className={styles.checkboxField}>
+			<input
+				{...buttonProps}
+				ref={ref}
+				id={undefined /* buttonProps.id should be assigned to the Radix checkbox instead */}
+				autoComplete="off"
+				className="sr-only"
+				aria-hidden
+				tabIndex={-1}
+				onFocus={() => buttonRef.current?.focus()}
```

Comment:
Conform will focus on the first invalid input on submit. This allow us to forward it to the button